### PR TITLE
🐛 Pin kindest/node images to known good versions in clusterctl upgrade tests

### PR DESCRIPTION
**Pull request by killianmuldoon**
Pin the kind image used in the clusterctl upgrade tests to a known good version which works with CAPD and KIND v0.19.0. These images are the ones suggested in the KIND v0.19.0 release at https://github.com/kubernetes-sigs/kind/releases/tag/v0.19.0

This should fix test failures related to #8815

### CODE REVIEW

**Comment by killianmuldoon** [comment]
/hold 

To revert the PR-Informing changes.

**Comment by killianmuldoon** [comment]
/retitle  🐛 Pin KIND nodes images to known good versions

File: test/e2e/data/infrastructure-docker/v1.3/bases/mp.yaml [hunk 1]
```diff
@@ -34,6 +34,8 @@ metadata:
   name: "${CLUSTER_NAME}-dmp-0"
 spec:
   template:
+      # NOTE: If the Kubernetes version is changed in `clusterctl_upgrade_test.go` the image and SHA must be updated here.
+      customImage: "kindest/node:v1.26.4@sha256:f4c0d87be03d6bea69f5e5dc0adb678bb498a190ee5c38422bf751541cebe92e"
```
Comment (killianmuldoon) [inline test/e2e/data/infrastructure-docker/v1.3/bases/mp.yaml:38]: In a bunch of those upgrade tests we have machinepool yamls, but I think we're not actually using them to generate cluster templates in most cases. For 1.0 we don't have those yamls at all.

Comment (killianmuldoon) [reply test/e2e/data/infrastructure-docker/v1.3/bases/mp.yaml:38]: Would it be better to re-include them or just get rid of them at all. As far as I can see they're only currently used in v0.4.0.

File: test/e2e/clusterctl_upgrade_test.go [hunk 2]
```diff
@@ -112,10 +114,11 @@ var _ = Describe("When testing clusterctl upgrades (v1.3=>current)", func() {
 			// try to deploy the latest version of our test-extension from docker.yaml.
 			InitWithRuntimeExtensionProviders: []string{},
 			InitWithProvidersContract:         "v1beta1",
-			InitWithKubernetesVersion:         "v1.26.4",
-			WorkloadKubernetesVersion:         "v1.26.4",
-			MgmtFlavor:                        "topology",
-			WorkloadFlavor:                    "",
+			// NOTE: If this version is changed here the image and SHA must also be updated in all DockerMachineTemplates in `test/data/infrastructure-docker/v0.4/bases.
```
[the remark this answers is not in the document]
Comment (killianmuldoon) [reply test/e2e/clusterctl_upgrade_test.go:117]: Will do

File: test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml [hunk 3]
```diff
@@ -196,30 +196,58 @@ spec:
...
-        kind: DockerMachineTemplate
-        matchResources:
-          controlPlane: true
-      jsonPatches:
-      - op: add
-        path: "/spec/template/spec/customImage"
-        valueFrom:
-          template: |
-            kindest/node:{{ .builtin.controlPlane.version | replace "+" "_" }}
+      - selector:
```
Comment (killianmuldoon) [inline test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml:199]: These patches should be reverted once we have kindest/node version resolution in CAPD.

**Comment by sbueringer** [comment]
Thank you very much!!

/lgtm
/approve

**Pull request merged**